### DISCUSSION TITLE
How to increase the warmup cache for HNSW + MMAP + Local SSD disk

### DISCUSSION THREAD

**Discussion by xiaobingxia-at** [milvus-io/milvus#38289] [Q&A and General discussion]
Hello, I'm using the HNSW + MMAP + Local SSD disk setup.
I observed that when doing the first search given specific collection, and corresponding filtering by partition key and clustering key, the first search would be 50-100ms. And following searches for same collection, partition key, clustering key will be <10ms.

But if the total data volume exceeds some level, when doing the intensive searches with a couple of threads, the search acceleration effect disappear. I assume that we run out of cache memory, and old entries got evicted from the cache. 

Is there anyway to increase the warmup memory cache size for HNSW + MMAP + Local SSD setup?

Thanks!

**Comment by xiaofan-luan** [comment]
> Hello, I'm using the HNSW + MMAP + Local SSD disk setup. I observed that when doing the first search given specific collection, and corresponding filtering by partition key and clustering key, the first search would be 50-100ms. And following searches for same collection, partition key, clustering key will be <10ms.
> 
> But if the total data volume exceeds some level, when doing the intensive searches with a couple of threads, the search acceleration effect disappear. I assume that we run out of cache memory, and old entries got evicted from the cache.
> 
> Is there anyway to increase the warmup memory cache size for HNSW + MMAP + Local SSD setup?
> 
> Thanks!

it is baed on your querynode memory limit. 
Milvus always try to fully utilize the mmory of you pod with page cache

**Reply by xiaobingxia-at** [reply]
My pod has enough memory, per dashboard, only 50% of memory is being used. However, first groups of searches are still slow. I hope it could use more memory.
